### PR TITLE
Feature/produce nuget for fs sonar runner core

### PR DESCRIPTION
**Pull request by jmecosta**
#75 update linter on sonar extensions

### CODE REVIEW

**Comment by milbrandt** [comment]
You need this as a nuget in some other place? Should this be permanently generated or only for working? If permanently I would propose to add it to the release notes.
Maybe we should make a minor release 3.1.1 anyway now that the deprecated api is removed.

**Comment by jmecosta** [comment]
Yeah, im kind of copy pasting the code on the visual studio extension, so
its best just to reuse this.

File: appveyor.yml [hunk 1]
```diff
@@ -30,9 +30,9 @@ dotnet_csproj:
   file: '**\*.csproj;**\*.props;**\*.fsproj;**\*.xml'
   version: '{version}'
   package_version: '{version}'
-  assembly_version: '{version}'
-  file_version: '{version}'
-  informational_version: '{version}'
+  assembly_version: '{version}.0'
```
Comment (milbrandt) [inline appveyor.yml]: shouldn't we change the version in line 8 accordingly and from here keep only reference?

**Comment by jmecosta** [comment]
Basically because we dont know if this number upgrades has changes on the
f# side i would rather have it created all the time.

**Comment by jmecosta** [comment]
I wouldnt mind, but then everything needs to be major.minor.buildnumber

**Comment by milbrandt** [comment]
> I wouldnt mind, but then everything needs to be major.minor.buildnumber

You wanted to reference on my review comment above? My proposal was be to define in line 8:

    version: 3.1.{build}.0

and then revert

    assembly_version: '{version}'
    file_version: '{version}'	
    informational_version: '{version}'

If ommitting the last `.0` helps, it's fine for me. 

I would prefer if all version stings are identical - so there can't be any misconceptions in case of bugs.

**Comment by jmecosta** [comment]
Yep, the last 0 then needs to go.

File: appveyor.yml (hunk 1 above)
Comment (milbrandt) [reply appveyor.yml]: Thanks for fixing it

**Review by milbrandt — CHANGES_REQUESTED** [review]
Please add some hint to the [RELEASE_NOTES.md](https://github.com/jmecsoftware/sonar-fsharp-plugin/blob/master/RELEASE_NOTES.md) that plugin versioning changed.

File: appveyor.yml [hunk 2]
```diff
@@ -5,7 +5,7 @@
 #---------------------------------#
 
 # version format
-version: 3.1.0.{build}
+version: 3.1.{build}
```
Comment (milbrandt) [inline appveyor.yml:8]: What is now displayed in the SonarQube Server Marketplace as version? Until now it was e.g. `3.1 (build 345)installed`

**Comment by jmecosta** [comment]
i think its more accurate since the version is 3.1.0.345 -> goes to  3.1
(build 345)
as 1.5.7 -> 1.5.7installed

but as far as i can see, i can find 3 types of verison

1. 1.1
2. 1.1.1
3. 1.3 (build x)
4. 1.4.3 (build x)

**Review by milbrandt — APPROVED** [review]

**Comment by milbrandt** [comment]
> Please add some hint to the [RELEASE_NOTES.md](https://github.com/jmecsoftware/sonar-fsharp-plugin/blob/master/RELEASE_NOTES.md) that plugin versioning changed.

I added now myself some release notes draft

**Comment by jmecosta** [comment]
Miss this one

**Comment by milbrandt** [comment]
> Miss this one

@jmecosta you will continue this PR? 
Do you intend to publish the new package to nuget.org? 
Shall we release a version because of the changed versioning scheme as soon as the PR is merged?

**Comment by jmecosta** [comment]
Yep, i will take care of it later today

**Comment by jmecosta** [comment]
@milbrandt since there are not functional changes i think we can skip this change and just release when we have more stuff coming in

**Comment by jmecosta** [comment]
nuget published https://www.nuget.org/packages/FsSonarRunnerCore/3.1.382

**Comment by milbrandt** [comment]
> @milbrandt since there are not functional changes i think we can skip this change and just release when we have more stuff coming in

@jmecosta you're right that this PR didn't change functionality. 
But since 3.1.0 in March there were several changes. And SonarQube wants an release at least every 6 month. All togerther I think we could  do a service release 3.1.382 to have the same versioning scheme for the released plugin as for nuget.
Publishing to nuget.org is not automated in the yaml pipeline as far as I saw?

**Comment by jmecosta** [comment]
Ok lets than do the release. I will check the auto deploy of nuget

**Comment by jmecosta** [comment]
@milbrandt should all be done now, 3.1.384 is now the only version in every place

**Comment by milbrandt** [comment]
👍

**Pull request merged**